### PR TITLE
liblwgeom: Avoid implicit type conversions

### PR DESCRIPTION
**Pull request by Algunenano**
WIP to avoid implicit type conversions (`-Wconversion`)

Continues with the spirit of https://github.com/postgis/postgis/pull/183:
- Standarizes `int32_t` as the type to store `srid`.
- Standarizes `uint32_t` as the type to manage `sizes` (some places used size_t, some int, some uint32_t).
- Changes pending `int` geometry iterators to `uint32_t` (missing from #183).
- [WIP] Use `int8_t` for boolean (currently mostly using `int`, `char`)

### CODE REVIEW

**Comment by Komzpa** [comment]
@Algunenano this seems to not work on gcc. 

```
make[1]: Entering directory `/home/travis/build/postgis/postgis/liblwgeom'
/bin/bash ../libtool --mode=compile gcc -I/usr/include   -I/usr/include/libxml2 -I/usr/include -DHAVE_SFCGAL    -g -O3 -mtune=generic -fno-omit-frame-pointer -Werror -Wall  -fPIC -DPIC  -Wall -Wmissing-prototypes -I/usr/include   -I/usr/include -DHAVE_SFCGAL -c -o g_serialized.lo g_serialized.c
libtool: compile:  gcc -I/usr/include -I/usr/include/libxml2 -I/usr/include -DHAVE_SFCGAL -g -O3 -mtune=generic -fno-omit-frame-pointer -Werror -Wall -fPIC -DPIC -Wall -Wmissing-prototypes -I/usr/include -I/usr/include -DHAVE_SFCGAL -c g_serialized.c  -fPIC -DPIC -o .libs/g_serialized.o
g_serialized.c:807:8: error: conflicting types for ‘gserialized_from_lwgeom_size’
 size_t gserialized_from_lwgeom_size(const LWGEOM *geom)
        ^
In file included from liblwgeom_internal.h:48:0,
                 from g_serialized.c:27:
liblwgeom.h:1961:17: note: previous declaration of ‘gserialized_from_lwgeom_size’ was here
 extern uint32_t gserialized_from_lwgeom_size(const LWGEOM *geom);
                 ^
make[1]: *** [g_serialized.lo] Error 1
make[1]: Leaving directory `/home/travis/build/postgis/postgis/liblwgeom'
make: *** [all] Error 1
[logbt] saw 'make' exit with code:2 (INT)
```

**Comment by Algunenano** [comment]
Yeah, sorry about that. It's done midway and I did not include yet the changes in `g_serialized.c` because I feared that I might have change too much there and wanted to test that what I had done until now worked also in the travis build. I failed to notice that not including those change would break compilation.

File: liblwgeom/lwgeom_geos.c [hunk 1]
```diff
@@ -112,7 +112,7 @@ GEOS2LWGEOM(const GEOSGeometry *geom, char want3d)
 {
 	int type = GEOSGeomTypeId(geom) ;
 	int hasZ;
-	int SRID = GEOSGetSRID(geom);
+	int SRID = (int32_t) GEOSGetSRID(geom);
```
Comment (Komzpa) [inline liblwgeom/lwgeom_geos.c]: This one looks strange.
Gets an int, casts to 32bit, casts to int.

Comment (Algunenano) [reply liblwgeom/lwgeom_geos.c]: Yes. I'll reset this file to its current status and review it again

**Comment by Komzpa** [comment]
Could you please add `-Wconversion` to the line in .travis.yml that has -Wall, so we can trace these too?

**Comment by Algunenano** [comment]
Closing this for now

**Pull request closed without merging**